scarb/tests/build.rs

```diff
@@ -635,3 +636,54 @@ fn workspace_as_dep() {
             )),
     );
 }
+
+#[test]
+fn can_define_edition() {
+    let t = TempDir::new().unwrap();
+    ProjectBuilder::start().edition("2023_01").build(&t);
+
+    let output = Scarb::quick_snapbox()
+        .arg("build")
+        .env("SCARB_LOG", "scarb=trace")
+        .current_dir(&t)
+        .output()
+        .expect("Failed to spawn command");
+
+    // Find project_config trace
+    for line in BufRead::split(output.stderr.as_slice(), b'\n') {
+        let line = line.expect("Failed to read line from stdout");
+        let line = String::from_utf8_lossy(&line);
+        if line.contains("project_config") {
+            // Actually match the output
+            snapbox::assert_matches(
+                r#"[..]ProjectConfig { base_path: "[..]", corelib: Some(Real("[..]")), content: ProjectConfigContent { crate_roots: OrderedHashMap({"pkg0": "[..]"}), crates_config: AllCratesConfig { global: SingleCrateConfig { edition: V2023_01 }, override_map: OrderedHashMap({"pkg0": SingleCrateConfig { edition: V2023_01 }, "core": SingleCrateConfig { edition: V2023_01 }}) } } }[..]"#,
+                line.as_ref(),
+            );
+            return;
+        }
+    }
+
+    panic!("Failed to find project_config in stderr");
+}
```

Comment:
I find this test overly complicated and prone to errors if we change something in trace logs. It's smart, I can't say it isn't, but I don't feel easy merging it.

Why not simply check if a piece of code using edition-specific feature is just compiling successfully?